Engine/CMakeLists.txt

```diff
@@ -171,102 +184,159 @@ set(VENDOR_PATH ${CMAKE_CURRENT_SOURCE_DIR}/vendor)
 # include dirs
 set(ENGINE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
-set(ENGINE_INCLUDE_DIRS
-${ENGINE_SOURCE_DIR}
-${VENDOR_PATH}/GLAD/include
-${VENDOR_PATH}/GLFW/include
-${VENDOR_PATH}/lia/include
-${VENDOR_PATH}/spdlog/include
-${VENDOR_PATH}/stb/include
-${VENDOR_PATH}/json/include
-${VENDOR_PATH}/freetype/include
-${VENDOR_PATH}/fmt/include
+# ------------------
+# Dependencies
+# ------------------
+FetchContent_Declare(
+        glfw
+        GIT_REPOSITORY "https://github.com/glfw/glfw.git"
+        GIT_TAG "3.4"
+)
+FetchContent_MakeAvailable(glfw)
+
+FetchContent_Declare(
+        lia
+        GIT_REPOSITORY "https://github.com/denyskryvytskyi/lia.git"
+        GIT_TAG "main"
+)
+FetchContent_MakeAvailable(lia)
+
+FetchContent_Declare(
+        spdlog
+        GIT_REPOSITORY "https://github.com/gabime/spdlog.git"
+        GIT_TAG "v1.14.1"
+)
+FetchContent_MakeAvailable(spdlog)
+
+FetchContent_Declare(json URL "https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz")
+FetchContent_MakeAvailable(json)
+
+FetchContent_Declare(
+        freetype
+        GIT_REPOSITORY "https://github.com/freetype/freetype.git"
+        GIT_TAG "VER-2-13-2"
+)
+FetchContent_MakeAvailable(freetype)
+
+FetchContent_Declare(
+        fmt
+        GIT_REPOSITORY "https://github.com/fmtlib/fmt.git"
+        GIT_TAG "10.2.1"
+)
+FetchContent_MakeAvailable(fmt)
+
+FetchContent_Declare(assimp
+GIT_REPOSITORY "https://github.com/assimp/assimp.git"
+GIT_TAG "v5.3.1")
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
+set(ASSIMP_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+set(ASSIMP_INJECT_DEBUG_POSTFIX OFF CACHE BOOL "" FORCE)
+set(ASSIMP_INSTALL OFF CACHE BOOL "" FORCE)
+set(ASSIMP_BUILD_ALL_EXPORTERS_BY_DEFAULT OFF CACHE BOOL "" FORCE)
+set(ASSIMP_BUILD_ZLIB OFF CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(assimp)
```

Comment:
Configuration for building Assimp. I explicitly disabled building of exporters as not sure you will need them right now. Zlib build also must be disabled to avoid conflicts with our zlib

Reply:
@MrOnlineCoder I have an issue while configuring the project. It can't find zlib while configuring assimp. Maybe you could help me.

![image](https://github.com/denyskryvytskyi/ElvenEngine/assets/25298585/58a6e6a4-ab20-496b-bfcc-33d8e69e8b1a)

Reply:
Hmm, interesting. I guess we'll have to build zlib strictly before assimp. I will push a fix for it